src/rust/engine/pantsd/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+#![deny(warnings)]
+// Enable all clippy lints except for many of the pedantic ones. It's a shame this needs to be copied and pasted across crates, but there doesn't appear to be a way to include inner attributes from a common source.
+#![deny(
+  clippy::all,
+  clippy::default_trait_access,
+  clippy::expl_impl_clone_on_copy,
+  clippy::if_not_else,
+  clippy::needless_continue,
+  clippy::unseparated_literal_suffix,
+  clippy::used_underscore_binding
+)]
+// It is often more clear to show that nothing is being moved.
+#![allow(clippy::match_ref_pats)]
+// Subjective style.
+#![allow(
+  clippy::len_without_is_empty,
+  clippy::redundant_field_names,
+  clippy::too_many_arguments
+)]
+// Default isn't as big a deal as people seem to think it is.
+#![allow(clippy::new_without_default, clippy::new_ret_no_self)]
+// Arc<Mutex> can be more clear than needing to grok Orderings:
+#![allow(clippy::mutex_atomic)]
+
+pub mod pantsd_testing;
+#[cfg(test)]
+mod pantsd_tests;
+
+use std::path::{Path, PathBuf};
+use std::{fmt, fs};
+
+use libc::pid_t;
+use log::debug;
+use options::{option_id, BuildRoot, OptionId, OptionParser, OptionType};
+use sha2::digest::Update;
+use sha2::{Digest, Sha256};
+use sysinfo::{ProcessExt, ProcessStatus, System, SystemExt};
+
+pub struct ConnectionSettings {
+  pub port: u16,
+  pub timeout_limit: f64,
+  pub dynamic_ui: bool,
+}
+
+impl ConnectionSettings {
+  pub fn new(port: u16) -> ConnectionSettings {
+    ConnectionSettings {
+      port,
+      timeout_limit: 60.0,
+      dynamic_ui: true,
+    }
+  }
+}
+
```

Comment:
This file was moved and heavily changed: looking at commits 2 and 3 separately may help (although they ended up slightly smeared together).